sys/syslog/backend.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2020 Continental
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_syslog
+ *
+ * @brief       Syslog backend definitions
+ *
+ * @{
+ * @file
+ *
+ * @author      Vincent Dupont <vincent.dupont@continental-its.com>
+ */
+
+#ifndef SYSLOG_BACKEND_H
+#define SYSLOG_BACKEND_H
+
+#include <stdint.h>
+#include <stdatomic.h>
+
+#include "thread.h"
+#include "time.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef CONFIG_SYSLOG_MAX_STRING_LEN
+#define CONFIG_SYSLOG_MAX_STRING_LEN    256
+#endif
+
+struct syslog_msg {
+    /* metadata */
+    uint8_t pri;
+    struct tm time;
```

Comment:
As above, IMO it would be better to use ztimer. A `ztimer_now_t` is also only 8 bytes in size (with `ztimer_now64` used), while `struct tm` is in between 36 and 44 Bytes on ARM. The backend could use a helper function to convert this to year, month, day, hour, minute and seconds.

It might be reasonable to add a `tm_t` type that contains only the reasonable fields using reasonable data times. IMO we should also consider updating the `periph_rtc` API to use that, so that driver authors don't need to worry about the day of the week, daylight saving settings, etc. in addition to saving same bytes for RAM. Maybe such a `tm_t` is better than `ztimer_now_t` for this use case.

Reply:
Unfortunately, there is no high-level API that takes advantage of RTC. The 'real' date is printed in the logs, so RTC is handy.
Though I agree, the full `struct tm` is a bit overkill and it may be too limiting to force the dependency to `periph_rtc`.
We could imagine having different implementations depending on if RTC is available I guess.